src/main/java/org/mars/rover/kata/MarsNavigator.java

```diff
@@ -1,60 +1,72 @@
 package org.mars.rover.kata;
 
-import org.mars.rover.kata.commands.CommandParser;
 import org.mars.rover.kata.entrydata.CommandSet;
 import org.mars.rover.kata.entrydata.RoverInstructions;
 import org.mars.rover.kata.entrydata.StdinProcessor;
+import org.mars.rover.kata.location.Coordinate;
+import org.mars.rover.kata.location.Grid;
+import org.mars.rover.kata.location.OccupiedArea;
+import org.mars.rover.kata.location.Position;
 
-import java.io.ByteArrayInputStream;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Scanner;
+import java.util.stream.IntStream;
 
 public class MarsNavigator {
     CommandSet providedInput;
 
-    ArrayList<ArrayList<Coordinate>> grid;
+    Grid grid;
 
     List<MarsRover> marsRovers;
 
     public MarsNavigator(CommandSet providedInput) {
         this.providedInput = providedInput;
-        this.grid = new ArrayList<>();
+        this.grid = Grid.createGrid(providedInput.gridX(), providedInput.gridY());
         this.marsRovers = new ArrayList<>();
     }
 
-
     public void processCommandSet() {
-        this.createGrid();
-        this.marsRovers = this.constructRovers(this.providedInput.roverInstructions()); // todo write test
-    }
-
-    protected void createGrid() {
-        int width = this.providedInput.gridX();
-        int height = this.providedInput.gridY();
-
-        for (int i = 0; i < width; i++) {
-            ArrayList<Coordinate> row = new ArrayList<>();
-
-            for (int j = 0; j < height; j++) {
-                row.add(new Coordinate(i, j));
-            }
-
-            grid.add(row);
-        }
+        this.marsRovers = this.constructRovers(this.providedInput.roverInstructions());
```

Comment:
Tu cały czas nie kumam, po co zapisujemy sparsowane komendy do `this.providedInput`? W sumie processing komend robota to takie przetwarzanie strumieniowe. Nie ma potrzeby komend nigdzie zapisywać, można je wykonywać w locie.

Reply:
W takim razie, uwaga do omówienia na naszym następnym spotkaniu - dane wejściowe zapisujemy dlatego, ponieważ jest przydatne do debugowania i łatwiej jest dokonywać na nich operacji (np. ustalając przeszkody). Ewentualna decyzję w tym względzie zostawiam na później :smiley: